megatron/core/models/mamba/mamba_model.py

```diff
@@ -129,8 +150,13 @@ def __init__(
             pg_collection=self.pg_collection,
         )
 
+        if self.mtp_process:
+            self.mtp = MultiTokenPredictionBlock(
+                config=self.config, spec=self.mtp_block_spec, vp_stage=vp_stage, mtp_hybrid_override_pattern=self.mtp_hybrid_override_pattern
+            )
```

Comment:
Why is MTP not another layer type that just gets instantiated in the block above? Why is it treated specially/differently?